Data/pubtator_to_xml.py

```diff
@@ -0,0 +1,180 @@
+    # Write out relevant tag
+    if tag["Type"] == "Gene":
+        annt.infons["NCBI Gene"] = tag["ID"]
+
+    elif tag["Type"] == "Species":
+        annt.infons["NCBI Species"] = tag["ID"]
+
+    elif "Mutation" in tag["Type"]:
+        annt.infons["tmVar"] = tag["ID"]
+
+    elif "SNP" in tag["Type"]:
+        annt.infons["tmVar"] = tag["ID"]
+
+    else:
+        # If there is no MESH ID for an annotation
+        if tag["ID"]:
+            # check to see if there are multiple mesh tags
+            if "|" in tag["ID"]:
+                # Write out each MESH id as own tag
+                for tag_num, ids in enumerate(tag["ID"].split("|")):
+                    # Some ids dont have the MESH:#### form so added case to that
+                    if ":" not in ids:
+                        annt.infons["MESH {}".format(tag_num)] = tag["ID"]
+                    else:
+                        term_type, term_id = ids.split(":")
+                        annt.infons["{} {}".format(term_type, tag_num)] = term_id
+            else:
+                # Some ids dont have the MESH:#### form so added case to that
+                if ":" in tag["ID"]:
+                    term_type, term_id = tag["ID"].split(":")
+                    annt.infons[term_type] = term_id
+                else:
+                    annt.infons["MESH"] = tag["ID"]
+        else:
+            annt.infons["MESH"] = "Unknown"
+
+    location = BioCLocation()
+    location.offset = str(tag["Start"])
+    location.length = str(len(tag["Term"]))
+    annt.locations.append(location)
+    annt.text = tag["Term"]
+    return annt
+
+
+def bioconcepts2pubtator_offsets(input_file):
+    file_lines = list()
+    with gzip.open(input_file, "rb") as f:
+        for line in f:
+            # Convert "illegal chracters" (i.e. < > &) in the main text
+            # into html entities
+            line = cgi.escape(line.rstrip()).encode("ascii", "xmlcharrefreplace")
+            if line:
+                file_lines.append(line)
+            else:
+                article = {}
+
+                # title
+                title_heading = file_lines[0].split('|')
+                title_len = len(title_heading[1])
+                article["Title"] = [title_heading[0], title_heading[1]]
+
+                # abstract
+                abstract_heading = file_lines[1]
+                article["Abstract"] = [abstract_heading[0], title_len, abstract_heading[1]]
+
+                # set up the csv reader
+
+                annts = csv.DictReader(file_lines[2:], fieldnames=['Document', 'Start', 'End', 'Term', 'Type', 'ID'], delimiter=str("\t"))
+                sorted_annts = sorted(annts, key=lambda x: x["Start"])
+                article["Title_Annot"] = filter(lambda x: x["Start"] < title_len, sorted_annts)
+                article["Abstract_Annot"] = filter(lambda x: x["Start"] > title_len, sorted_annts)
+
+                yield article
+                file_lines = list()
+
```

Comment:
Let's add `assert not file_lines` outside loop to make sure we're not missing anything